### DISCUSSION TITLE
Dependant queries

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#14137] [Q&A]
Hello,

Thank you very much for the great library !

I am having a difficult time to create a single transaction that would write in a specific record of a table, an information located in another table. 

So this would be considered as dependant transactions (read then write). Is there a way to do that in a single transaction ? 
Ideal solution to me would keep the use of data proxy  ( seems that this preview preview feature " interactivetransaction" is not working with dataproxy)

Thank you so much for you support !

**Comment by austincrim** [comment]
Hey @laurent512 👋,

While you are correct that `interactiveTransactions` are not yet available to use with the Data Proxy, what you are trying to accomplish might be possible with a plain (non-interactive) `prisma.$transaction` call.

Could you provide some sample code on what you are trying to accomplish?

**Comment by laurent512** [comment]
Hi @austincrim,

I have 3 relevant models here, Transaction, PayrollRow and Payroll. 

My two kind of customers send money between each others by issuing Transactions (from Voucher, to Shop).

After each month, I must compute the fees that must be charged for all the transaction issued. 
So I look at all the transactions done during that period, and compute the fees per transactions that must be charged and put that result into PayrollRow. 
Then I package all these PayrollRow in one Payroll that will be send to relevant parties. 
Payroll can be seen as a table, which contains many rows (PayrollRow), as well as the summary value of those (the field amountEurToPay).

I know having amountEurToPay in Payroll could be considered as redundancy (that could be obtained from aggregating data from its related PayrollRow) , but for auditing purposes I really want to have a summary value in the Payroll 


```
model Transaction {
  id          Int       @id @default(autoincrement())
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt
  cancelledAt DateTime?
  from        Voucher   @relation(fields: [fromId], references: [id])
  fromId      Int
  to          Shop      @relation(fields: [toId], references: [id])
  toId        Int
  payroll     Payroll?  @relation(fields: [payrollId], references: [id])
  payrollId   Int?
  amountEur   Decimal   @default(0) @db.Decimal(10, 2)
}

model PayrollRow {
  id            Int       @id @default(autoincrement())
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @updatedAt
  txId          Int
  txCreatedAt   DateTime  @default(now())
  txCancelledAt DateTime?
  txAmountEur   Decimal   @default(0) @db.Decimal(10, 2)
  txAbsFeesEur  Decimal   @default(0) @db.Decimal(10, 2)
  txRelFeesEur  Decimal   @default(0) @db.Decimal(10, 2)
  paybackEur    Decimal   @default(0) @db.Decimal(10, 2)
  payroll       Payroll   @relation(fields: [payrollId], references: [id])
  payrollId     Int
}

model Payroll {
  id             Int           @id @default(autoincrement())
  createdAt      DateTime      @default(now())
  updatedAt      DateTime      @updatedAt
  status         PayrollStatus @default(pending)
  recipient      Shop          @relation(fields: [recipientId], references: [id])
  recipientId    Int
  fileURL        String        @default("") // file URL should point to pdf containing everything that should be build once at creation time ; this will act as official document
  transactions   Transaction[]
  amountEurToPay Decimal       @default(0) @db.Decimal(10, 2)
  payrollRow     PayrollRow[]
  payment        Payment?      @relation(fields: [paymentId], references: [id])
  paymentId      Int?
}
```

**Comment by austincrim** [comment]
It looks like you would need interactive transactions for that.

We do plan on bringing support for interactive transactions to the Data Proxy, watch our [roadmap](https://pris.ly/roadmap) for updates 👍.

**Reply by laurent512** [reply]
thank you for your reply ! 
It seems that it is not already included in the roadmap right ? 

Also, when I use queries that are not supported by dataproxy, should I completely disable it or prisma will bypass it by itself automatically?

**Reply by austincrim** [reply]
> It seems that it is not already included in the roadmap right ?

I'll ask about this, but we definitely plan on adding support.

> Also, when I use queries that are not supported by dataproxy, should I completely disable it or prisma will bypass it by itself automatically?

You will need to disable the Data Proxy for now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
